### PR TITLE
 Refactor and clean up e2e framework utils, this patch handles test/e2e/framework/psp_util.go file

### PR DESCRIPTION
**Pull request by WanLinghao**
/kind cleanup

**What this PR does / why we need it**:
 Refactor and clean up e2e framework utils, this patch handles test/e2e/framework/perf_util.go file
ref:#76206

```release-note
NONE
```

### CODE REVIEW

**Comment by WanLinghao** [comment]
/hold
wait for https://github.com/kubernetes/kubernetes/pull/77536 merged

**Comment by fejta** [comment]
Can you create OWNERS files for the psp dir please?
/assign @timothysc

**Review by timothysc — CHANGES_REQUESTED** [review]
/approve 

but you need add the generated files and OWNERs files.

**Comment by andrewsykim** [comment]
Overall lgtm, can you rebase please?

**Comment by alejandrox1** [comment]
Hi @WanLinghao any updates on this?
/assign

**Comment by WanLinghao** [comment]
@alejandrox1 Sorry for the delay, I have checked this, it is not ready to merge. We must take care of `ExpectNoError` functions before continue. Or it could cause package import cycle.

**Comment by alejandrox1** [comment]
> @alejandrox1 Sorry for the delay, I have checked this, it is not ready to merge. We must take care of `ExpectNoError` functions before continue. Or it could cause package import cycle.

Sorry for the delay as well, there is actually been some debate about how to handle `framework.ExpectNoError()`.

To get in the same page, `framework.ExpectNoError()` wraps a call to
```go
framework.ExpectNoErrorWithOffset(1, err, explain...)
```

`framework.ExpectNoErrorWithOffset()` itself being a wrapper for
```go
gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred(), explain...)}
```
The first argument to `gomega.ExpectWithOffset()` represents the offset in the call stack.

For example, if you have a test, and you perform certain actions in your test through helper functions and an assertion in one of these helper function fails the line numbers provided by Gomega will point you to the line in the helper function that failed instead of showing you the line in your test that failed.

`framework.ExpectNoError()` will, by default, call `gomega.ExpectWithOffset()` with an offset equal to 1, which means that a failed assertion will be logged for the caller of the method where `gomega.ExpectWithOffset()` is used.

`framework.ExpectNoError()` was then implemented with the aim of using it within helper functions in e2e tests.
This way, if an assertion failed we would obtain information about the specific step within the e2e test that failed and would avoid having to backtrace what part of what test made a call to the given function.

With that in mind, in order to refactor code from the framework into its own package within the framework (i.e., e2e/framework/psp), we need to replace calls to `framework.ExpectNoError()` with either `gomega.Expect()` or `gomega.ExectWithOffset()`.
The simplest way would be then to replace a `framework.ExpectNoError()` with the following:
```go
if err != nil {
	e2elog.Logf("Unexpected error occurred: %v", err)
}
gomega.ExpectWithOffset(1+offset, err).NotTo(gomega.HaveOccurred(), explain...)
```

Which replaces the call to `framework.ExpectNoError()` with the [code that it ends up calling)https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1371).

**Comment by WanLinghao** [comment]
@alejandrox1 Hi, this patch is ready to review

**Review by fejta — COMMENTED** [review]
/lgtm
/approve
/hold
Do we want an owners file for framework/psp?

**Comment by WanLinghao** [comment]
@fejta  Hi, I updated the patch to fix a golint error.

File: test/e2e/framework/psp/psp.go [hunk 1]
```diff
@@ -149,8 +152,16 @@ func createPrivilegedPSPBinding(f *Framework, namespace string) {
...
 			schema.GroupResource{Group: "extensions", Resource: "podsecuritypolicies"}, true))
 	}
 }
+
+// ExpectNoError is a copy from the same name function in file test/e2e/framework.go
+func ExpectNoError(err error, explain ...interface{}) {
+	if err != nil {
+		e2elog.Logf("Unexpected error occurred: %v", err)
+	}
+	gomega.ExpectWithOffset(2, err).NotTo(gomega.HaveOccurred(), explain...)
```
Comment (alejandrox1) [inline test/e2e/framework/psp/psp.go]: the `ExpectNoError` from the framework calls `gomega.ExpectWithOffset(1, err)` not `gomega.ExpectWithOffset(2, err)`, correct? If so, we should change this.
This patch changes the current behavior. Which is not wrong but we should add it in a separate PR and only focus on refactoring for this on. This way we can revert if something starts failing in CI.

/lgtm cancel

**Review by alejandrox1 — CHANGES_REQUESTED** [review]

**Comment by alejandrox1** [comment]
> Do we want an owners file for framework/psp?

I believe the consensus thus far has to keep all e2e framework subpkgs under the same ownership.
/priority backlog

File: test/e2e/framework/psp/psp.go (hunk 1 above)
Comment (WanLinghao) [reply test/e2e/framework/psp/psp.go]: @alejandrox1  Do you mean we should change  `gomega.ExpectWithOffset(2, err).NotTo(gomega.HaveOccurred(), explain...)` to` gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred(), explain...)`? But I believe the original definition should be  `gomega.ExpectWithOffset(2, err).NotTo(gomega.HaveOccurred(), explain...)`

Comment (alejandrox1) [reply test/e2e/framework/psp/psp.go]: Sorry, you are completely right!
Given https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1321
a `framework.NoExpectError` is equivalent to `gomega.ExpectWithOffset(2, err).NotTo(gomega.HaveOccurred(), explain...)`.

/lgtm
/hold cancel

**Review by alejandrox1 — APPROVED** [review]

**Pull request merged**